cli/util/uploader.go

```diff
@@ -396,7 +396,13 @@ func updateIndexFiles(ctx *cli.Context, p *pool.Pool, containerID cid.ID, accoun
 				err := retry(func() error {
 					var errSearchIndex error
 					objIDs, errSearchIndex = neofs.ObjectSearch(ctx.Context, p, account.PrivateKey(), containerID.String(), prm)
-					return errSearchIndex
+					if errSearchIndex != nil {
+						return errSearchIndex
+					}
+					if len(objIDs) != searchBatchSize {
```

Comment:
It's not always `searchBatchSize`, it must work with `searchBatchSize` set to `37`, remember? You have `j` and `end` to define the expected number of blocks.